From c7eb0046f9179d9d7fd451fa6c41affee010e3e6 Mon Sep 17 00:00:00 2001
From: oxalica <oxalicc@pm.me>
Date: Mon, 16 Dec 2024 15:05:18 -0500
Subject: [PATCH] Report preferred buffer size for window screencasting

So client can use a buffer in matching size, taking display scaling into
account, to avoid re-scaling the buffer.
---
 src/plugins/screencast/windowscreencastsource.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/plugins/screencast/windowscreencastsource.cpp b/src/plugins/screencast/windowscreencastsource.cpp
index f2a1432e4d..2b2c6ed36d 100644
--- a/src/plugins/screencast/windowscreencastsource.cpp
+++ b/src/plugins/screencast/windowscreencastsource.cpp
@@ -49,12 +49,12 @@ quint32 WindowScreenCastSource::drmFormat() const
 
 QSize WindowScreenCastSource::textureSize() const
 {
-    return m_window->clientGeometry().size().toSize();
+    return m_window->clientGeometry().size().toSize() * devicePixelRatio();
 }
 
 qreal WindowScreenCastSource::devicePixelRatio() const
 {
-    return 1.0;
+    return m_window->preferredBufferScale();
 }
 
 void WindowScreenCastSource::render(QImage *target)
-- 
2.47.0

